linux/LinuxProcess.c

```diff
@@ -189,6 +198,279 @@ void LinuxProcess_printDelay(float delay_percent, char* buffer, int n) {
 }
 #endif
 
+/*
+Note that when colorizing a basename with the comm prefix, the entire basename
+(not just the comm prefix) is colorized for better readability, and it is
+implicit that only upto (TASK_COMM_LEN - 1) could be comm
+*/
+static inline bool findCommInCmdline(const char *comm, char *cmdline, int cmdlineBasenameOffset, int *pCommStart, int *pCommEnd) {
+   /* Try to find procComm in tokenized cmdline - this might in rare cases
+    * mis-identify a string or fail, if comm or cmdline had been unsuitably
+    * modified by the process */
+   bool lastToken = false;
+   char *token, *tokenBase;
+
+   for (token = cmdline + cmdlineBasenameOffset; *token; ) {
+      for (tokenBase = token; *token && *token != '\n'; ++token) {
+         if (*token == '/')
+            tokenBase = token + 1;
+      }
+      if (*token)    /* NUL terminate at delimiter for strncmp */
+         *token = 0;
+      else
+         lastToken = true;
+
+      if (strncmp(tokenBase, comm, TASK_COMM_LEN - 1) == 0) {
+         if (!lastToken)   /* restore the delimiter */
+            *token = '\n';
+         *pCommStart = tokenBase - cmdline;
+         *pCommEnd = token - cmdline;
+         return true;
+      }
+      if (!lastToken) {
+         /* restore the delimiter and skip repeats */
+         *token = '\n';
+         while (*++token == '\n')
+            ;
```

Comment:
Avoid the empty loop body. Better `do { token++; } while('\n' == *token);`.
Also the missing buffer length guarding against accidental reads behind the end of the buffer somewhat irritates me (Overall missing in this routine).

Reply:
Standard idiom in string processing, but still will address to conform to htop coding style.